.gitmodules

```diff
@@ -4,3 +4,13 @@
 [submodule "aclk/aclk-schemas"]
 	path = aclk/aclk-schemas
 	url = https://github.com/netdata/aclk-schemas.git
+[submodule "ml/kmeans/dlib"]
+	path = ml/kmeans/dlib
+	url = https://github.com/davisking/dlib.git
+	shallow = true
+	ignore = dirty
+	shallow = true
+[submodule "ml/json"]
```

Comment:
not a fan of adding another json lib in parallel to one we already have but i get it (ease of use from C++)

Reply:
I understand, on the other hand the JSON conversion logic takes exactly 1 line of C++ code. 

If we reach the conclusion that this is not something we want to do (and everything else in the PR looks good), I can add follow-up commits to rely on the C library instead.

Reply:
Well I am of preferring not having duplicate libs but that preference is not above all... depends on other stuff like how much more effort would be to use JSONC...

I will let other team members tell their opinion but don´t worry will not cry if we decide to keep it.